platforms/core-execution/build-cache-local/src/integTest/groovy/org/gradle/caching/local/internal/AbstractBuildCacheCleanupIntegrationTest.groovy

```diff
@@ -65,24 +65,7 @@ abstract class AbstractBuildCacheCleanupIntegrationTest extends AbstractIntegrat
                 description = "Generates a 1MB file"
             }
         """
-    }
-
-    def withDeprecatedBuildCacheRetentionInDays(long period) {
-        settingsFile << """
-            buildCache {
-                local {
-                    removeUnusedEntriesAfterDays = ${period}
-                }
-            }
-        """
-    }
-
-    def expectRetentionMethodDeprecationWarning() {
-        executer.expectDocumentedDeprecationWarning(
-            "The DirectoryBuildCache.removeEntriesAfterDays property has been deprecated. " +
-                "This is scheduled to be removed in Gradle 9.0. " +
-                "Consult the upgrading guide for further information: https://docs.gradle.org/current/userguide/upgrading_version_8.html#directory_build_cache_retention_deprecated"
-        )
+        requireOwnGradleUserHomeDir("Modifying cache settings in Gradle user home")
```

Comment:
❓ Why didn't we need this before?

Reply:
Every test that sets `gradleUserHome` did that manually. via `initializeHome()`. But when changing tests I found it a hard way that forgetting about that and setting cleanup strategy to common integration tests can make other tests behave in a funny way. 

So for future self I just added this to `setup()` to avoid problems in the future.